editor/editor.tsx

```diff
@@ -235,6 +235,20 @@ function Editor(props: any) {
         setHg(undefined);
     }, [demo]);
 
+    /**
+     * Convert relative CSV data URLs to absolute URLs.
+     * (e.g., './example.csv' => 'https://gist.githubusercontent.com/{urlGist}/raw/example.csv')
+     */
+    function resolveRelativeCsvUrls(spec: string) {
+        const newSpec = JSON.parse(spec);
+        gosling.traverseTracksAndViews(newSpec as gosling.GoslingSpec, (tv: any) => {
+            if (tv.data && tv.data.type === 'csv' && tv.data.url.indexOf('./') === 0) {
+                tv.data.url = tv.data.url.replace('./', `https://gist.githubusercontent.com/${urlGist}/raw/`);
+            }
+        });
+        return stringify(newSpec);
+    }
+
```

Comment:
I wonder if we could make this an even more reusable:

```typescript
/**
 * Convert relative CSV data URLs to absolute URLs.
 * (e.g., './example.csv' => 'https://gist.githubusercontent.com/{urlGist}/raw/example.csv')
 */
function resolveRelativeCsvUrls(spec: string, importMeta: URL) {
	const newSpec = JSON.parse(spec);
	// https://regex101.com/r/l87Q5q/1
	const relativePathRegex = /^[.\/]|^\.[.\/]|^\.\.[^\/]/
	gosling.traverseTracksAndViews(newSpec as gosling.GoslingSpec, (tv: any) => {
		if (tv.data&& tv.data.type === 'csv' && relativePathRegex.test(tv.data.url)) {
			tv.data.url = new URL(tv.data.url, importMeta).href;
		}
	});
	return stringify(newSpec);
}
```

and then:

```typescript
    const specUrl = new URL(`https://gist.githubusercontent.com/${gist}/raw/${dataFile}`);
    const whenCode = fetch(specUrl.href).then(async response =>
        response.status === 200 ? resolveRelativeCsvUrls(await response.text(), specUrl) : null
    );
```

the second argument to `URL` is a [`base`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) to resolve a path relative to.